.github/workflows/publish.yml

```diff
@@ -0,0 +1,29 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    branches:
+      - 'main'
+
+jobs:
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Setup Nodejs
+        uses: actions/setup-node@v1
+        with:
+          node-version: 16
+          always-auth: true
+          registry-url: 'https://registry.npmjs.org'
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
+      - name: Login NPM
+        run: npm login
+      - name: Build and Publish
+        run: cd js && yarn && yarn run publish
```

Comment:
I would suggest splitting yarn install and publish so you can provide
```
        env:
          NODE_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
```
to publish step and in that case you don't need `npm login` step